src/mapping.ts

```diff
@@ -1,410 +1,477 @@
+  let contract = fetchERC1155(event.address.toHexString());
+  let operator = fetchAccount(event.params.operator.toHexString());
+  let from = fetchAccount(event.params.from.toHexString());
+  let to = fetchAccount(event.params.to.toHexString());
 
-  let underlyingDayData = updateTokenDayData(underlyingAsset, event);
-  underlyingDayData.volume = underlyingDayData.volume.plus(underlyingAmount);
-  underlyingDayData.volumeUSD = underlyingDayData.volumeUSD.plus(
-    underlyingValueUSD
+  registerTransfer(
+    event,
+    "",
+    contract,
+    operator,
+    from,
+    to,
+    event.params.id,
+    event.params.amount
   );
-  underlyingDayData.save();
 
-  updateTokenDayData(exerciseAsset, event);
+  /**
+   * Valorem Extension
+   * Checks if to/from addresses are not one of Valorem's or the Zero Address
+   * If neither are, records the amounts transferred in daily metrics
+   */
+  // get params
+  const fromAddress = event.params.from.toHexString();
+  const toAddress = event.params.to.toHexString();
+  const tokenId = event.params.id;
+  const amount = event.params.amount;
+
+  // check transfer isn't an OSE Write/Exercise/Redeem event (mint/OSE is receiver/burn)
+  const transferIsExternal =
+    fromAddress != ZERO_ADDRESS &&
+    toAddress != ZERO_ADDRESS &&
+    toAddress != event.address.toHexString();
+
+  if (transferIsExternal) {
+    // bind to OptionSettlementEngine to call view functions
+    const ose = OSEContract.bind(Address.fromBytes(event.address));
+    // get type of Token
+    const tokenType = ose.tokenType(tokenId);
+
+    handleERC1155TransferMetrics(tokenId, amount, event, tokenType);
+  }
 }
 
-export function handleOptionsWritten(event: OptionsWritten): void {
-  let contract = fetchERC1155(event.address);
-  let token = fetchERC1155Token(contract, event.params.claimId);
-  token.claim = event.params.claimId.toString();
-  token.type = 2;
-  token.save();
-
-  let option = Option.load(event.params.optionId.toString())!;
-  // should never be null. if option doesnt exist, somethings wrong with handleNewOptionType
-  // if (option == null) {
-  //   option = new Option(event.params.optionId.toString());
-  //   option.save();
-  // }
-
-  const underlyingAssetAddress = option.underlyingAsset
-    ? option.underlyingAsset
-    : ZERO_ADDRESS;
-
-  const underlyingPriceUSD = getTokenPriceUSD(underlyingAssetAddress as string);
+// https://github.com/OpenZeppelin/openzeppelin-subgraphs
+export function handleTransferBatch(event: TransferBatchEvent): void {
+  const contract = fetchERC1155(event.address.toHexString());
+  const operator = fetchAccount(event.params.operator.toHexString());
+  const from = fetchAccount(event.params.from.toHexString());
+  const to = fetchAccount(event.params.to.toHexString());
+  const ids = event.params.ids;
+  const amounts = event.params.amounts;
+  // If this equality doesn't hold (some devs actually don't follow the ERC specifications) then we just can't make
```

Comment:
Not sure we need to keep this salty OZ comment in =)

Reply:
HA I agree.

Reply:
removed https://github.com/valorem-labs-inc/valorem-subgraph/commit/c09a1c3b1e5fee3131700cb87bbe0fe0d9804827